mongo/integration/unified/collection_data.go

```diff
@@ -20,20 +20,41 @@ import (
 )
 
 type collectionData struct {
-	DatabaseName   string     `bson:"databaseName"`
-	CollectionName string     `bson:"collectionName"`
-	Documents      []bson.Raw `bson:"documents"`
+	DatabaseName   string                 `bson:"databaseName"`
+	CollectionName string                 `bson:"collectionName"`
+	Documents      []bson.Raw             `bson:"documents"`
+	Options        *collectionDataOptions `bson:"collectionOptions"`
+}
+
+type collectionDataOptions struct {
+	Capped      *bool  `bson:"capped"`
+	SizeInBytes *int64 `bson:"size"`
 }
 
 // createCollection configures the collection represented by the receiver using the internal client. This function
-// first drops the collection and then creates it and inserts the seed data if needed.
+// first drops the collection and then creates it with specified options (if any) and inserts the seed data if needed.
 func (c *collectionData) createCollection(ctx context.Context) error {
 	db := mtest.GlobalClient().Database(c.DatabaseName)
 	coll := db.Collection(c.CollectionName, options.Collection().SetWriteConcern(mtest.MajorityWc))
```

Comment:
```suggestion
	db := mtest.GlobalClient().Database(c.DatabaseName, options.Database().SetWriteConcern(mtest.MajorityWc))
	coll := db.Collection(c.CollectionName, options.Collection())
```

The `CreateCollection` operation requires majority write concern.

> All setup operations MUST use the internal MongoClient and a "majority" write concern

Reply:
Great point; set the majority write concern on the `Database` instead of the `Collection` to have majority write concern for the `createCollection` operations.